rust/crd/src/lib.rs

```diff
@@ -637,10 +641,13 @@ pub struct PodRef {
 }
 
 impl PodRef {
-    pub fn fqdn(&self) -> String {
+    pub fn fqdn(&self, cluster_info: &KubernetesClusterInfo) -> String {
         format!(
-            "{}.{}.{}.svc.cluster.local",
-            self.pod_name, self.role_group_service_name, self.namespace
+            "{}.{}.{}.svc.{}",
+            self.pod_name,
+            self.role_group_service_name,
+            self.namespace,
+            cluster_info.cluster_domain
```

Comment:
nit: We should avoid unnamed interpolations (as they then depend on order which is fragile).

```suggestion
            "{pod_name}.{service_name}.{namespace}.svc.{cluster_domain}",
            pod_name = self.pod_name,
            service_name = self.role_group_service_name,
            namespace = self.namespace,
            cluster_domain = cluster_info.cluster_domain
```

You're welcome to leave it for now, as there are other occurrences anyway.

Reply:
Was expecting this :)